etna/transforms/decomposition/stl.py

```diff
@@ -120,16 +129,27 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
 
         Returns
         -------
-        result: pd.DataFrame
+        :
             Dataframe with extracted features
         """
         result = df
-        if self.fit_results is not None:
-            season_trend = self.fit_results.get_prediction(
-                start=df[self.in_column].first_valid_index(), end=df[self.in_column].last_valid_index()
-            ).predicted_mean
-        else:
+        if self.fit_results is None:
             raise ValueError("Transform is not fitted! Fit the Transform before calling transform method.")
+
+        first_valid_index = df[self.in_column].first_valid_index()
+        last_valid_index = df[self.in_column].last_valid_index()
+
+        if pd.api.types.is_integer_dtype(df.index):
+            start = first_valid_index - self._first_int_index
+            end = last_valid_index - self._first_int_index
+            if start < 0:
+                raise ValueError("Transform can't work on integer timestamp before training data!")
+
+            season_trend = self.fit_results.get_prediction(start=start, end=end).predicted_mean
```

Comment:
As I explained higher, holt doesn't work without index. I don't understand why arima works, but holt doesn't, but there was an error.

Here we use integer index indices (like `iloc`) because it is easier than getting the correct timestamp from fake index and still makes it possible to get predictions even in the future.

Reply:
May you add notes about holt here